File: backend/src/main/java/de/hkwh/backend/datatransferobjects/WallboxDTO.java
package de.hkwh.backend.datatransferobjects;

import javax.validation.constraints.NotBlank;

public record WallboxDTO(
        @NotBlank long p_id,
        @NotBlank long h_id,
        @NotBlank String spotName,
        @NotBlank boolean isFree,
        int chargingSpeed
) {
}

File: frontend/src/model/WallboxDTO.ts
import {VehicleScheduleDTO} from "./VehicleScheduleDTO";

export interface WallboxDTO {
    p_id: number;
    h_id:  number;
    spotName: string;
    isFree: boolean;
    chargingSpeed: number
    vehicles_scheduled: VehicleScheduleDTO[];
}